test/route.js

```diff
@@ -633,23 +633,23 @@ describe('Router', function () {
         it('should work in a named parameter', function (done) {
           var cb = after(2, done)
           var router = new Router()
-          var route = router.route('/:foo(*)')
+          var route = router.route('/:foo(.*)')
           var server = createServer(router)
 
           route.all(sendParams)
 
           request(server)
           .get('/bar')
-          .expect(200, {'0': 'bar', 'foo': 'bar'}, cb)
+          .expect(200, {'foo': 'bar'}, cb)
```

Comment:
Hmm, I did not do this change.  I wonder what I did.  I will look into why this is changed and fix it before we merge this.

Reply:
I'm not sure if we need to keep the old behavior if the new behavior is better, but we need to clearly explain how someone looking at their routers will know that the result of req.param is going to change

Reply:
Ok, I will add this to the guide and leave the change here.  I do think the named argument is what I would expect from the route defined in this test.

Reply:
done